### DISCUSSION TITLE
General Installation Process to Follow

### DISCUSSION THREAD

**Discussion by ma1120** [firefly-iii/firefly-iii#6394] [General]
Hi! I'm not sure if this exists, but is there any recommended documentation for installation that go through all the .env variables (other than the comments in the file itself)? I'm running into a couple of issues that I think are a result of a misunderstanding of the variables, and am having some trouble finding examples/additional documentation

**Comment by JC5** [comment]
Could you share the ones you're having trouble with?

**Reply by ma1120** [reply]
I think these are all pretty basic questions, I just couldn't find any documentation (fireflyiii specific or not) to clear them up. The three big ones are:

For the .env database credentials, should that match the docker-compose.yml environment user/password/database (using mariadb, so mysql but hosted locally if I understand that correctly)? That currently returns an error when trying to build with the .yml.

Can SSL be added after the db is created? Or does it have to be established before the docker-compose file is run?

What is the purpose behind running a reverse proxy for fireflyiii? I have a very basic understanding of what that entails, but am not seeing how it applies for this case.

Again, I think pretty low level questions, but I haven't been able to find any resolution so far. Thank you very much for any help!

**Reply by timaschew** [reply]
The last two questions are not really related to firefly. I guess that's why you didn't find it in the docs.

1. Yes the credentials in your .env file must match wiith the values in the docker-compose.yml 
You say, it returns an error. Please, **always** send details about the error, otherwise nobody is able to help you. I had trouble with the mardiadb image and and older docker version. My error was related to a bug with the time sync or so. I've downgraded the version and then it worked, I'm using now this version in the docker-composey.yml `image: mariadb:10.8.2`

2. I don't think SSL for MySQL make sense if it's running on the same host and exposes no connections to the external networks.

3. With a reverse proxy:
 - you can handle https without touching/configurinig the app (firefly)
 - you can remap the ports, firefly runs by default on port 8080, but you can map it to 80 or 443 (with https)
 - it can help you to handle virtual hosts for you, so that you don't need to use an IP and a custom port, but any DNS firefly.local or firefly.your-domain.tld
 - and do lot of other stuff: https://en.wikipedia.org/wiki/Nginx#Features
